mesonbuild/dependencies/base.py

```diff
@@ -373,25 +411,23 @@ def get_as_shared(self, recursive: bool) -> InternalDependency:
         return new_dep
 
 class ExternalDependency(Dependency):
-    def __init__(self, type_name: DependencyTypeName, environment: 'Environment', kwargs: T.Dict[str, T.Any], language: T.Optional[str] = None):
+    def __init__(self, type_name: DependencyTypeName, environment: 'Environment', kwargs: DependencyKWs, language: T.Optional[str] = None):
         Dependency.__init__(self, type_name, kwargs)
         self.env = environment
         self.name = type_name # default
         self.is_found = False
         self.language = language
         version_reqs = kwargs.get('version', None)
-        if isinstance(version_reqs, str):
-            version_reqs = [version_reqs]
-        self.version_reqs: T.Optional[T.List[str]] = version_reqs
+        self.version_reqs = version_reqs
         self.required = kwargs.get('required', True)
         self.silent = kwargs.get('silent', False)
         static = kwargs.get('static')
         if static is None:
-            static = self.env.coredata.get_option(OptionKey('prefer_static'))
-        self.static = T.cast('bool', static)
+            static = T.cast('bool', self.env.coredata.get_option(OptionKey('prefer_static')))
+        self.static = static
```

Comment:
It does not matter much, but unless there is something subtle going on that I don't see, this hunk seem to simply change on which line the cast is done and could be either squashed with a patch a few commits down the stack, or simply suppressed.

Reply:
Right, so mypy complains it's a useless cast because if `static` comes from `kwargs`, we now know what type it is, but if it comes from options then it's `int | bool | str | list[str]`. I honestly feel like that's a mypy bug, it really should be complaining that `static` is changing types if you don't cast.